src/validation.cpp

```diff
@@ -350,21 +350,29 @@ void CChainState::MaybeUpdateMempoolForReorg(
     // the disconnectpool that were added back and cleans up the mempool state.
     m_mempool->UpdateTransactionsFromBlock(vHashUpdate);
 
-    const auto check_final_and_mature = [this, flags=STANDARD_LOCKTIME_VERIFY_FLAGS](CTxMemPool::txiter it)
+    // Check whether the transaction is still final and, if it spends a coinbase output, mature.
+    // Update the entry's cached LockPoints if needed. If this returns true, the tx would be invalid
```

Comment:
> Update the entry's cached LockPoints if needed

This is a bit vague with the current code. The lock points may or may not be updated if this function returns `true` and will be updated for sure if it returns `false`. I think it is better to update them in all cases or change this comment to say "Update the entry's cached LockPoints if false is returned (and also in some cases if true is returned :disappointed:)".